src/io/json.cpp

```diff
@@ -627,27 +596,19 @@ namespace fc
               os << "null";
               return;
          case variant::int64_type:
-         {
-              int64_t i = v.as_int64();
               if( format == json::stringify_large_ints_and_doubles &&
-                  i > 0xffffffff )
+                  ( v.as_int64() > 0xffffffff || v.as_int64() < -int64_t(0xffffffff) ) )
```

Comment:
Actually I think the range should be `[-0x7fffffff, 0x7fffffff]`

Reply:
AFAIK the reason for this is the number representation in some scripting languages like JavaScript, which is based on IEEE doubles with 52 bits of precision. 2^32-1 should be safe, as well as -(2^32-1).

Reply:
... unless doing bit-shifting.
https://stackoverflow.com/questions/307179/what-is-javascripts-highest-integer-value-that-a-number-can-go-to-without-losin

Reply:
The string representation will not enable bit shifting on large integers either.

Reply:
> The string representation will not enable bit shifting on large integers either.

True, but the point is, if it represents as a number, it should enable bit shifting, so 2^31 -1 is the max safe number.